### PR TITLE
proxy 14-15 (setup eslint && refactor -> TS)

### PR DESCRIPTION
**Pull request by parazit-mc**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,9 +2,10 @@
   "name": "nasa-node-1",
   "version": "1.0.0",
   "description": "test nasa api",
-  "main": "src/index.js",
+  "main": "src/index.ts",
   "scripts": {
-    "dev": "nodemon src/index.js"
+    "dev": "nodemon src/index.ts",
```
Comment (Mikhail1992) [inline package.json:7]: does hot reload works?

File: src/index.ts [hunk 2]
```diff
@@ -32,11 +32,11 @@ app.use(router);
...
-        console.log(`listen to port ${PORT}`);
-    }
+const server = app.listen(PORT, () => {
+    console.log(`Server is running on port ${PORT}`);
+});
+
+server.on('error', (error: Error) => {
+    console.error('Server startup error:', error);
+    Sentry.captureException(error);
 });
```
Comment (Mikhail1992) [inline src/index.ts:42]: handle "graceful shutdown" please

File: src/services/meteorService.ts [hunk 3]
```diff
@@ -1,40 +1,54 @@
...
+import { Request } from 'express';
+import { DateTimeFormatOptions } from 'intl';
+import { Meteor } from "../models/meteorModel";
 
-async function getMeteors(req) {
+interface MeteorNearby {
+    [key: string]: Meteor[];
+}
+
+export async function getMeteors(req: Request): Promise<Meteor[] | undefined> {
```
Comment (Mikhail1992) [inline src/services/meteorService.ts:11]: please don't provide req for services. Provide only required data

File: src/validators/validator.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+export function validateRequest(schema: Schema, property: keyof Request) {
+    return (req: Request, res: Response, next: NextFunction) => {
+        const { error } = schema.validate(req[property]);
+        if (error) {
+            const errorMessage = property === 'body' ? 'Request body malformed' : 'Path param contains errors';
+            const validationError = new RequestValidationError(errorMessage);
+            validationError.statusCode = 400;
+            validationError.message = error.details[0].message;
+            Sentry.captureException(validationError);
+            return res.status(400).json({ status: 400, message: errorMessage, error: error.details[0].message });
```
Comment (Mikhail1992) [inline src/validators/validator.ts:23]: you can throw exception here

File: package.json (hunk 1 above)
Comment (parazit-mc) [reply package.json:7]: as expected

Comment (parazit-mc) [reply package.json:7]: > does hot reload works?

as expected

File: src/index.ts (hunk 2 above)
Comment (parazit-mc) [reply src/index.ts:51]: > handle "graceful shutdown" please

done

File: src/services/meteorService.ts (hunk 3 above)
Comment (parazit-mc) [reply src/services/meteorService.ts]: > please don't provide req for services. Provide only required data

done

File: src/validators/validator.ts (hunk 4 above)
Comment (parazit-mc) [reply src/validators/validator.ts]: > you can throw exception here

done

**Pull request merged**